src/components/header/Header.tsx

```diff
@@ -26,35 +24,40 @@ const Header = () => {
 
     //change id of the film for film page
     if (filmIndex > -1) {
-      if (films) dispatch(changeId(films[filmIndex].id));
+      //clean input field
+      setInputValue("");
+      if (films)
+        //jump to the film page
+        history.push(`/film-page/${films[filmIndex].id}`);
     }
-
-    //clean input field
-    setInputValue("");
-
-    //jump to the film page
-    history.push("/film_page");
   };
+
   return (
-    <>
+    <div className="header">
       <h1>Film World</h1>
-      <div className="mini_form">
+      <div className="mini-form">
         <Link to="/">
-          <button className="home_button">Home</button>
+          <button className="home-button">Home</button>
         </Link>
-        <div className="input_container">
+        <div className="input-container">
           <input
+            list="films"
             className="input"
             value={inputValue}
             placeholder="search for a film..."
             onChange={(event) => setInputValue(event.target.value)}
           />
-          <button onClick={handleSearch} className="search_button">
+          <datalist id="films">
+            {films?.map((film) => (
+              <option value={film.title} key={film.id} />
+            ))}
+          </datalist>
+          <button onClick={handleSearch} className="search-button">

```

Comment:
can't find handleSearch function, search doesn't seem to be working. What if on search button icon click, we take inputValue, and then try to find a movie with title containing that string in our store. If found we redirect to that movie page (for simplicity purposes to the first occurence if there are more than one), or to not found page